### PR TITLE
🐛 Fix IMAGE_ID issue in examples

### PR DESCRIPTION
**Pull request by ncdc**
**What this PR does / why we need it**:
Remove the ${IMAGE_ID} placeholder from the examples, as it's only used
for e2e conformance testing, and it's broken for example usage.

Use kustomize patches to handle the IMAGE_ID for e2e testing instead.

Cherry-pick of #1336 for `master`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by ncdc** [comment]
@dims does this look ok to you?

**Comment by ncdc** [comment]
/hold
```
# Create control plane machine.
kubectl \
	--kubeconfig=$(kind get kubeconfig-path --name="clusterapi") \
	create -f examples/_out/controlplane.yaml
error: no objects passed to create
```

grrr

**Comment by vincepri** [comment]
FWIW, I have seen this failure in other PRs as well

**Comment by ncdc** [comment]
The `error: no objects passed to create` was my fault. I fixed that.

The most recent failure was the control plane never came up, but in theory the Machines and AWSMachines were all created.

**Comment by ncdc** [comment]
Found some things in the ec2 instance's journal. Not sure if these are relevant or red herrings:
- `secret \"webhook-server-cert\" not found"` (failed volume mount)
- secret \"cert-manager-webhook-tls\" not found" (failed volume mount)

**Comment by ncdc** [comment]
Hmm, seeing that in master too, in https://storage.googleapis.com/kubernetes-jenkins/logs/ci-cluster-api-provider-aws-make-conformance-master/1192488049464840192/artifacts/logs/clusterapi-control-plane/journal.log

Maybe we should merge this and work on fixing whatever is keeping the apiserver from running separately?

**Comment by ncdc** [comment]
AHA!
```
controller-runtime/controller "msg"="Reconciler error" "error"="failed to create AWSMachine instance: failed to run instance: InvalidAMIID.Malformed: Invalid id: \"${IMAGE_ID}\" (expecting \"ami-...\")\n\tstatus code: 400, request id: be6ed54d-e26a-4f56-bd18-2bb0e6d0b53f"  "controller"="awsmachine" "request"={"Namespace":"default","Name":"test1-controlplane-0"}
```

So maybe I need to do some more work.

/lgtm cancel

**Comment by ncdc** [comment]
FWIW, this is not because of this PR. It's also happening in master: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-cluster-api-provider-aws-make-conformance-master/1192488049464840192/artifacts/logs/clusterapi-control-plane/pods/capa-system_capa-controller-manager-84778d978f-6rm79_4ddf2dda-a935-49f2-bbb6-33040c413c94/manager/0.log

File: Makefile [hunk 1]
```diff
@@ -293,8 +293,10 @@ release-notes: $(RELEASE_NOTES)
 ## Development
 ## --------------------------------------
 
+# NOTE: do not add 'generate-exmaples' as a prerequisite of this target. It will break e2e conformance testing.
 .PHONY: create-cluster-management
-create-cluster-management: $(CLUSTERCTL) generate-examples ## Create a development Kubernetes cluster on AWS in a KIND management cluster.
+create-cluster-management: $(CLUSTERCTL) ## Create a development Kubernetes cluster on AWS in a KIND management cluster.
+	@if [[ ! -f examples/_out/cert-manager.yaml ]]; then echo "Examples are missing. Run 'make generate-examples' first."; exit 1; fi
```
Comment (dims) [inline Makefile]: Legit! +1 :)

**Comment by ncdc** [comment]
Working!
/hold cancel

**Pull request merged**